tests/integration/conftest.py

```diff
@@ -164,6 +164,17 @@ def batch_for_datasource(
     yield _batch_setup_for_datasource.make_batch()
 
 
+@pytest.fixture
+def asset_for_datasource(
+    _batch_setup_for_datasource: BatchTestSetup,
+) -> Generator[DataAsset, None, None]:
+    """Fixture that yields an asset for a specific data source type.
+    This must be used in conjunction with `indirect=True` to defer execution
+    """
+    set_context(_batch_setup_for_datasource.context)
```

Comment:
Why is this necessary?

Reply:
Because we cache the setup objects and may reuse them across tests, we might have created/set a different context since this context was created/set.